extensions/amp-audio/1.0/component.js

```diff
@@ -0,0 +1,195 @@
+    src,
+    title,
+    toggleFallback,
+    validateMediaMetadata,
+    ...rest
+  } = props;
+
+  /** @public {boolean} */
+  const isPlaying = useRef(false);
+
+  /**
+   * Checks if the function is allowed to be called
+   * @return {boolean}
+   */
+  const isInvocationValid_ = useCallback(() => {
+    if (!audioRef) {
+      return false;
+    }
+
+    if (isStoryDescendant_ && isStoryDescendant_()) {
+      console /*OK*/
+        .warn(
+          '<amp-story> elements do not support actions on <amp-audio> elements'
+        );
+      return false;
+    }
+    return true;
+  }, [isStoryDescendant_]);
+
+  /**
+   * Prepares Media Metadata
+   */
+  const metaData = useMemo(() => {
+    return {
+      title,
+      artist,
+      album,
+      artwork: [{src: artwork}],
+    };
+  }, [title, artist, album, artwork]);
+
+  /**
+   * Plays audio callback
+   */
+  const playCallback = useCallback(() => {
+    triggerAnalyticsEvent(audioRef.current, 'audio-play');
+
+    if (!isInvocationValid_()) {
+      return;
+    }
+    audioRef.current.play();
+    isPlaying.current = true;
+  }, [isPlaying, isInvocationValid_]);
+
+  /**
+   * Pauses audio callback
+   */
+  const pauseCallback = useCallback(() => {
+    triggerAnalyticsEvent(audioRef.current, 'audio-pause');
+
+    if (!isInvocationValid_()) {
+      return;
+    }
+    audioRef.current.pause();
+    isPlaying.current = false;
+  }, [isPlaying, isInvocationValid_]);
+
+  /**
+   * Updates media session for current window/tab
+   */
+  const audioPlaying = useCallback(() => {
+    const win = audioRef.current?.ownerDocument?.defaultView;
+    const element = audioRef.current;
+
+    if (validateMediaMetadata) {
+      validateMediaMetadata(element, metaData);
+    }
+
+    setMediaSession(win, metaData, playCallback, pauseCallback);
+  }, [metaData, validateMediaMetadata, playCallback, pauseCallback]);
+
+  useEffect(() => {
+    if (!audioRef.current.play) {
+      if (toggleFallback) {
+        toggleFallback(true);
+      }
+      return;
+    }
+
+    const unlistenPlaying = listen(audioRef.current, 'playing', () =>
+      audioPlaying()
+    );
+
+    if (propagateAttributes) {
+      propagateAttributes(
+        [
+          'aria-describedby',
+          'aria-label',
+          'aria-labelledby',
+          'autoplay',
+          'controlsList',
+          'loop',
+          'muted',
+          'preload',
+          'src',
+        ],
+        audioRef.current
+      );
+    }
+
+    // Execute at unlayout
+    return () => {
+      unlistenPlaying();
+    };
+  }, [audioPlaying, children, propagateAttributes, toggleFallback]);
+
+  /** Audio Component - API Functions */
+  useImperativeHandle(
+    ref,
+    () =>
+      /** @type {!AudioDef.AudioApi} */ ({
+        play: () => playCallback(),
+        pause: () => pauseCallback(),
+        isPlaying: () => isPlaying.current,
+      }),
+    [playCallback, pauseCallback]
+  );
+
+  return (
+    <audio
```

Comment:
Please wrap this with `ContainWrapper`, similar to `amp-video`:
https://github.com/ampproject/amphtml/blob/5a26e0a2d0e9ffa2d88c586d7f79d43997c6f240/extensions/amp-video/1.0/component.js#L262-L269